File: src/client/xnative/lion.xdroid/Control/SettingPagerFragment.cs
using Android.App;
using Android.Content;
using Android.OS;
using Android.Util;
using Android.Views;
using Android.Widget;
using Lion.XDroid;
using Lion.XDroid.Libs;
using Lion.XDroid.Popup;
using Lion.XDroid.Setting;
using Lion.XDroid.Type;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;

namespace Lion.XDroid.Control
{
    public class SettingPagerFragment : Android.Support.V4.App.Fragment, IPagerFragment
    {
        private const int NAME_CHANGE_POPUP = 1;
        private const int MAX_NUMBER_POPUP = 2;
        private const int FIX_NUMBER_POPUP = 3;

        public event EventHandler<RefreshEventArgs> Refresh;

        View rootView;

        List<UserInfo> user_infos;

        public SettingPagerFragment()
        {
        }

        public override void OnCreate(Bundle savedInstanceState)
        {
            //Log.Debug(this.GetType().Name, $"OnCreate");
            base.OnCreate(savedInstanceState);

            user_infos = new List<UserInfo>();
        }

        public override View OnCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
        {
            //Log.Debug(this.GetType().Name, $"OnCreateView ({((ViewPager)container).CurrentItem})");

            this.rootView = inflater.Inflate(Resource.Layout.fragment_setting, container, false);

            this.InitSize(this.rootView);
            this.InitEvent(this.rootView);
            this.InitData(this.rootView);

            return this.rootView;
        }

        public override void OnActivityResult(int requestCode, int resultCode, Intent data)
        {
            var app_cache = new AppPreferences(this.Context);
            var _token = app_cache.UserTokenKey;

            var _login_id = app_cache.UserLoginId;
            var _user_info = user_infos
                                .Where(u => u.login_id == _login_id)
                                .SingleOrDefault();

            switch (requestCode)
            {
                case NAME_CHANGE_POPUP:
                    if (resultCode == (int)Result.Ok && data != null)
                    {
                        var _old_name = data.GetStringExtra("old_name");
                        var _new_name = data.GetStringExtra("new_name");
                        if (_old_name != _new_name)
                        {
                            _user_info.loginName = _new_name;
                            this.UpdateUserInfor(_token, _user_info);
                        }
                    }
                    break;

                case MAX_NUMBER_POPUP:
                    if (resultCode == (int)Result.Ok && data != null)
                    {
                        _user_info.maxSelectNumber = (short)data.GetIntExtra("number", 0);
                        this.UpdateUserInfor(_token, _user_info);
                    }
                    break;

                case FIX_NUMBER_POPUP:
                    if (resultCode == (int)Result.Ok && data != null)
                    {
                        var _number = data.GetIntExtra("number", 0);

                        var _exist_check = false;
                        if ((_user_info.digit1 != 0 && _user_info.digit1 == _number)
                            || (_user_info.digit2 != 0 && _user_info.digit2 == _number)
                            || (_user_info.digit3 != 0 && _user_info.digit3 == _number))
                            _exist_check = true;

                        if (_exist_check == false)
                        {
                            if (_number == 0)
                            {
                                _user_info.digit3 = 0;
                                _user_info.digit2 = 0;
                                _user_info.digit1 = 0;
                            }
                            else
                            {
                                if (_user_info.digit1 == 0)
                                    _user_info.digit1 = (short)_number;
                                else if (_user_info.digit2 == 0)
                                    _user_info.digit2 = (short)_number;
                                else
                                    _user_info.digit3 = (short)_number;
                            }

                            this.UpdateUserInfor(_token, _user_info);
                        }
                        else
                            AppDialog.SNG.Alert(this.Context, "동일 번호를 선택 하셨습니다.");
                    }
                    break;
            }
        }

        public void Initialize()
        {
            this.InitData(this.rootView);
        }

        private void InitSize(View view)
        {
            var iv_label_name = view.FindViewById<ImageView>(Resource.Id.iv_label_name);
            var iv_label_email = view.FindViewById<ImageView>(Resource.Id.iv_label_email);
            var iv_label_max = view.FindViewById<ImageView>(Resource.Id.iv_label_max);
            var iv_label_fix = view.FindViewById<ImageView>(Resource.Id.iv_label_fix);
            var iv_label_pw = view.FindViewById<ImageView>(Resource.Id.iv_label_pw);
            var iv_label_push = view.FindViewById<ImageView>(Resource.Id.iv_label_push);

            iv_label_name.LayoutParameters = AppCommon.SNG.ResizeDP(this.Context, iv_label_name.LayoutParameters);
            iv_label_email.LayoutParameters = AppCommon.SNG.ResizeDP(this.Context, iv_label_email.LayoutParameters);
            iv_label_max.LayoutParameters = AppCommon.SNG.ResizeDP(this.Context, iv_label_max.LayoutParameters);
            iv_label_fix.LayoutParameters = AppCommon.SNG.ResizeDP(this.Context, iv_label_fix.LayoutParameters);
            iv_label_pw.LayoutParameters = AppCommon.SNG.ResizeDP(this.Context, iv_label_pw.LayoutParameters);
            iv_label_push.LayoutParameters = AppCommon.SNG.ResizeDP(this.Context, iv_label_push.LayoutParameters);

            var btn_logout = view.FindViewById<Button>(Resource.Id.btn_logout);
            var btn_leave = view.FindViewById<Button>(Resource.Id.btn_leave);

            btn_logout.LayoutParameters = AppCommon.SNG.ResizeDP(this.Context, btn_logout.LayoutParameters);
            btn_leave.LayoutParameters = AppCommon.SNG.ResizeDP(this.Context, btn_leave.LayoutParameters);
        }

        private void InitEvent(View view)
        {
            var iv_label_name = view.FindViewById<ImageView>(Resource.Id.iv_label_name);
            var iv_label_email = view.FindViewById<ImageView>(Resource.Id.iv_label_email);
            var iv_label_max = view.FindViewById<ImageView>(Resource.Id.iv_label_max);
            var iv_label_fix = view.FindViewById<ImageView>(Resource.Id.iv_label_fix);
            var iv_label_pw = view.FindViewById<ImageView>(Resource.Id.iv_label_pw);
            var iv_label_push = view.FindViewById<ImageView>(Resource.Id.iv_label_push);
            var btn_logout = view.FindViewById<Button>(Resource.Id.btn_logout);
            var btn_leave = view.FindViewById<Button>(Resource.Id.btn_leave);

            iv_label_name.Touch += ListenerTouchIV;
            iv_label_email.Touch += ListenerTouchIV;
            iv_label_max.Touch += ListenerTouchIV;
            iv_label_fix.Touch += ListenerTouchIV;
            iv_label_pw.Touch += ListenerTouchIV;
            iv_label_push.Touch += ListenerTouchIV;

            iv_label_name.Click += ListenerClickIV;
            iv_label_email.Click += ListenerClickIV;
            iv_label_max.Click += ListenerClickIV;
            iv_label_fix.Click += ListenerClickIV;
            iv_label_pw.Click += ListenerClickIV;
            iv_label_push.Click += ListenerClickIV;

            btn_logout.Touch += ListenerTouchBT;
            btn_leave.Touch += ListenerTouchBT;
            btn_logout.Click += ListenerClickBT;
            btn_leave.Click += ListenerClickBT;
        }

        private void InitData(View view)
        {
            var app_cache = new AppPreferences(this.Context);
            var _token = app_cache.UserTokenKey;
            var _login_id = app_cache.UserLoginId;

            var _user_info = user_infos
                                .Where(u => u.login_id == _login_id)
                                .SingleOrDefault();

            if (_user_info == null)
            {
                this.GetCount(_token, _user_info);
                this.GetUserInfo(_token, _login_id);
            }
            else
            {
                var _interval = Convert.ToInt32(Resource.String.push_read_interval_minute);
                if (_user_info.last_push_count_read.AddMinutes(_interval) < DateTime.Now)
                    this.GetCount(_token, _user_info);

                DisplayView(_user_info);
            }
        }

        private void ListenerClickBT(object sender, System.EventArgs e)
        {
            var app_cache = new AppPreferences(this.Context);

            var view = (Button)sender;
            switch (view.Id)
            {
                case Resource.Id.btn_logout:
                    app_cache.UserTokenKey = "";
                    Refresh?.Invoke(this, new RefreshEventArgs(true));
                    break;

                case Resource.Id.btn_leave:
                    AppDialog.SNG.CancelAlert(this.Context, "모든 정보가 삭제됩니다.\n정말로 탈퇴하시겠습니까?",
                        (s1, e1) =>
                        {
                            var _token = app_cache.UserTokenKey;
                            LeaveMember(_token);
                        });
                    break;
            }
        }

        private void ListenerTouchBT(object sender, View.TouchEventArgs e)
        {
            var view = (Button)sender;

            if (e.Event.Action == MotionEventActions.Down)
            {
                view.Background.SetColorFilter(0x50000000);
                view.Invalidate();
            }
            else if (e.Event.Action == MotionEventActions.Up)
            {
                view.Background.SetColorFilter(0x00000000);
                view.Invalidate();
            }
            else if (e.Event.Action == MotionEventActions.Cancel)
            {
                view.Background.SetColorFilter(0x00000000);
                view.Invalidate();
            }

            e.Handled = false;
        }

        private void ListenerClickIV(object sender, System.EventArgs e)
        {
            var _intent = (Intent)null;

            var view = (ImageView)sender;
            switch (view.Id)
            {
                case Resource.Id.iv_label_name:
                    var tv_name = this.rootView.FindViewById<TextView>(Resource.Id.tv_name);

                    _intent = new Intent(this.Context, typeof(SettingNameActivity));
                    _intent.PutExtra("old_name", tv_name.Text);
                    StartActivityForResult(_intent, NAME_CHANGE_POPUP);
                    break;

                case Resource.Id.iv_label_email:
                    _intent = new Intent(this.Context, typeof(SettingEmailActivity));
                    StartActivity(_intent);
                    break;

                case Resource.Id.iv_label_max:
                    _intent = new Intent(this.Context, typeof(NumberPopupActivity));
                    _intent.PutExtra("min", 5);
                    _intent.PutExtra("max", 100);
                    _intent.PutExtra("reverse", false);
                    StartActivityForResult(_intent, MAX_NUMBER_POPUP);
                    break;

                case Resource.Id.iv_label_fix:
                    _intent = new Intent(this.Context, typeof(NumberPopupActivity));
                    _intent.PutExtra("min", 0);
                    _intent.PutExtra("max", 45);
                    _intent.PutExtra("reverse", false);
                    StartActivityForResult(_intent, FIX_NUMBER_POPUP);
                    break;

                case Resource.Id.iv_label_pw:
                    _intent = new Intent(this.Context, typeof(SettingPwActivity));
                    StartActivity(_intent);
                    break;

                case Resource.Id.iv_label_push:
                    PutCount(0);

                    _intent = new Intent(this.Context, typeof(SettingPushActivity));
                    StartActivity(_intent);
                    break;
            }
        }

        private void ListenerTouchIV(object sender, View.TouchEventArgs e)
        {
            var view = (ImageView)sender;

            if (e.Event.Action == MotionEventActions.Down)
            {
                view.Background.SetColorFilter(0x30000000);
            }
            else if (e.Event.Action == MotionEventActions.Up)
            {
                view.Background.SetColorFilter(0x00000000);
            }
            else if (e.Event.Action == MotionEventActions.Cancel)
            {
                view.Background.SetColorFilter(0x00000000);
                view.Invalidate();
            }

            e.Handled = false;
        }

        private void GetUserInfo(string user_token, string login_id)
        {
            try
            {
                this.Loading(true);

                var api = new ApiAsnycTask(this.Context, GetString(Resource.String.api_url) + "user/GetUserInfor", user_token);
                api.Execute();

                api.SendFinish += (s, e) =>
                {
                    var _result = JsonConvert.DeserializeObject<ApiResult<UserInfo>>(e.Json);
                    if (_result != null)
                    {
                        if (_result.success == true)
                        {
                            _result.result.login_id = login_id;

                            user_infos.RemoveAll(x => x.login_id == login_id);
                            user_infos.Add(_result.result);

                            DisplayView(_result.result);
                        }
                        else
                            AppDialog.SNG.Alert(this.Context, _result.message);
                    }

                    this.Loading(false);
                };
            }
            catch (Java.Lang.Exception ex)
            {
                Log.Error(this.GetType().Name, ex.Message);
            }
        }

        private void UpdateUserInfor(string user_token, UserInfo user_info)
        {
            try
            {
                this.Loading(true);

                var _parameter = new Dictionary<string, string>();
                {
                    _parameter.Add("login_name", user_info.loginName);
                    _parameter.Add("max_select_number", user_info.maxSelectNumber.ToString());
                    _parameter.Add("digit1", user_info.digit1.ToString());
                    _parameter.Add("digit2", user_info.digit2.ToString());
                    _parameter.Add("digit3", user_info.digit3.ToString());
                }

                var api = new ApiAsnycTask(this.Context, GetString(Resource.String.api_url) + "user/UpdateUserInfor", user_token);
                api.Execute(_parameter);

                api.SendFinish += (s, e) =>
                {
                    var _result = JsonConvert.DeserializeObject<ApiResult<string>>(e.Json);
                    if (_result != null)
                    {
                        if (_result.success == true)
                        {
                            AppDialog.SNG.Alert(this.Context, "변경되었습니다.",
                            (s2, e2) =>
                            {
                                Initialize();
                            });
                        }
                        else
                            AppDialog.SNG.Alert(this.Context, _result.message);
                    }

                    this.Loading(false);
                };
            }
            catch (Java.Lang.Exception ex)
            {
                Log.Error(this.GetType().Name, ex.Message);
            }
        }

        private void PutCount(int push_count)
        {
            var app_cache = new AppPreferences(this.Context);
            var _login_id = app_cache.UserLoginId;
            var _user_info = user_infos
                                .Where(u => u.login_id == _login_id)
                                .SingleOrDefault();

            if (_user_info != null)
            {
                _user_info.last_push_count_read = DateTime.Now;
                _user_info.push_count = push_count;
            }
        }

        private void GetCount(string user_token, UserInfo user_info)
        {
            try
            {
                var api = new ApiAsnycTask(this.Context, GetString(Resource.String.api_url) + "notify/GetCount", user_token);
                api.Execute();

                api.SendFinish += (s, e) =>
                {
                    var _result = JsonConvert.DeserializeObject<ApiResult<string>>(e.Json);
                    if (_result != null)
                    {
                        if (_result.success == true)
                        {
                            //var tv_push = this.rootView.FindViewById<TextView>(Resource.Id.tv_push);
                            //tv_push.SetText(_result.result);

                            PutCount(Convert.ToInt32(_result.result));
                        }
                        else
                            AppDialog.SNG.Alert(this.Context, _result.message);
                    }
                };
            }
            catch (Java.Lang.Exception ex)
            {
                Log.Error(this.GetType().Name, ex.Message);
            }
        }

        private void LeaveMember(string user_token)
        {
            try
            {
                this.Loading(true);

                var api = new ApiAsnycTask(this.Context, GetString(Resource.String.api_url) + "user/LeaveMember", user_token);
                api.Execute();

                api.SendFinish += (s, e) =>
                {
                    var _result = JsonConvert.DeserializeObject<ApiResult<string>>(e.Json);
                    if (_result != null)
                    {
                        if (_result.success == true)
                        {
                            var app_cache = new AppPreferences(this.Context);
                            app_cache.UserTokenKey = "";

                            Refresh?.Invoke(this, new RefreshEventArgs(true));
                        }
                        else
                            AppDialog.SNG.Alert(this.Context, _result.message);
                    }

                    this.Loading(false);
                };
            }
            catch (Java.Lang.Exception ex)
            {
                Log.Error(this.GetType().Name, ex.Message);
            }
        }

        private void DisplayView(UserInfo user_info)
        {
            var tv_name = this.rootView.FindViewById<TextView>(Resource.Id.tv_name);
            tv_name.SetText(user_info.loginName);

            var tv_email = this.rootView.FindViewById<TextView>(Resource.Id.tv_email);
            tv_email.SetText(user_info.emailAddress);

            var tv_max = this.rootView.FindViewById<TextView>(Resource.Id.tv_max);
            tv_max.SetText(user_info.maxSelectNumber.ToString());

            var _digit_list = new int[]
                            {
                                user_info.digit1, user_info.digit2, user_info.digit3
                            };

            var _fix_number = "";
            foreach (var _digit in _digit_list)
            {
                if (_digit > 0)
                {
                    if (_fix_number.Length > 0)
                        _fix_number += ", ";

                    _fix_number += _digit.ToString();
                }
            }

            var tv_fix = this.rootView.FindViewById<TextView>(Resource.Id.tv_fix);
            tv_fix.SetText(_fix_number);

            var tv_push = this.rootView.FindViewById<TextView>(Resource.Id.tv_push);
            tv_push.SetText(user_info.push_count.ToString());
        }

        private void Loading(bool view)
        {
            var btn_loading = this.rootView.FindViewById<Button>(Resource.Id.btn_loading);
            var pb_loading = this.rootView.FindViewById<ProgressBar>(Resource.Id.pb_loading);

            if (view)
            {
                btn_loading.Visibility = ViewStates.Visible;
                pb_loading.Visibility = ViewStates.Visible;
            }
            else
            {
                btn_loading.Visibility = ViewStates.Invisible;
                pb_loading.Visibility = ViewStates.Invisible;
            }
        }
    }
}

File: src/client/xcode/Lion.Droid/app/src/main/java/com/odinsoftware/lottolion/control/SettingPagerFragment.java
package com.odinsoftware.lottolion.control;

import android.app.Activity;
import android.content.Intent;
import android.graphics.PorterDuff;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.ProgressBar;
import android.widget.TextView;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import com.odinsoftware.lottolion.R;
import com.odinsoftware.lottolion.popup.NumberPopupActivity;
import com.odinsoftware.lottolion.service.ApiAsnycTask;
import com.odinsoftware.lottolion.service.LLAlert;
import com.odinsoftware.lottolion.service.LLCommon;
import com.odinsoftware.lottolion.service.LLSharedPreferences;
import com.odinsoftware.lottolion.setting.SettingEmailActivity;
import com.odinsoftware.lottolion.setting.SettingPushActivity;
import com.odinsoftware.lottolion.setting.SettingPwActivity;
import com.odinsoftware.lottolion.type.ApiResult;
import com.odinsoftware.lottolion.type.UserInfo;

import java.util.ArrayList;
import java.util.Collections;
import java.util.HashMap;

/**
 * Created by limyg on 2017-04-26.
 */
public class SettingPagerFragment extends Fragment {

    private final int MAX = 1;
    private final int FIX = 2;
    View rootView;
    UserInfo userInfo;

    public interface Refresh{
        void onRefresh();
    }
    public Refresh refresh;
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        rootView = inflater.inflate(R.layout.fragment_setting, container, false);
        InitSize(rootView);
        InitEvent(rootView);
        InitData();

        return rootView;
    }

    private void InitData() {
        String token = LLSharedPreferences.LL(getContext()).getSaveSession();
        if(LLCommon.Get().CheckExpired(token) == false) {
            GetUserInfo(token);
            GetCount(token);
        }
    }

    public void Init(){
        InitData();
    }

    private void InitEvent(View view) {
        ImageView iv_label_email = (ImageView)view.findViewById(R.id.iv_label_email);
        ImageView iv_label_max = (ImageView)view.findViewById(R.id.iv_label_max);
        ImageView iv_label_fix = (ImageView)view.findViewById(R.id.iv_label_fix);
        ImageView iv_label_pw = (ImageView)view.findViewById(R.id.iv_label_pw);
        ImageView iv_label_push = (ImageView)view.findViewById(R.id.iv_label_push);
        Button btn_logout = (Button)view.findViewById(R.id.btn_logout);
        Button btn_leave = (Button)view.findViewById(R.id.btn_leave);

        iv_label_email.setOnTouchListener(listener_touch_iv);
        iv_label_email.setOnClickListener(listener_click_iv);
        iv_label_max.setOnTouchListener(listener_touch_iv);
        iv_label_max.setOnClickListener(listener_click_iv);
        iv_label_fix.setOnTouchListener(listener_touch_iv);
        iv_label_fix.setOnClickListener(listener_click_iv);
        iv_label_pw.setOnTouchListener(listener_touch_iv);
        iv_label_pw.setOnClickListener(listener_click_iv);
        iv_label_push.setOnTouchListener(listener_touch_iv);
        iv_label_push.setOnClickListener(listener_click_iv);

        btn_logout.setOnTouchListener(listener_touch_btn);
        btn_logout.setOnClickListener(listener_click_btn);
        btn_leave.setOnTouchListener(listener_touch_btn);
        btn_leave.setOnClickListener(listener_click_btn);
    }

    private void InitSize(View view) {
        ImageView iv_label_email = (ImageView)view.findViewById(R.id.iv_label_email);
        ImageView iv_label_max = (ImageView)view.findViewById(R.id.iv_label_max);
        ImageView iv_label_fix = (ImageView)view.findViewById(R.id.iv_label_fix);
        ImageView iv_label_pw = (ImageView)view.findViewById(R.id.iv_label_pw);
        ImageView iv_label_push = (ImageView)view.findViewById(R.id.iv_label_push);
        Button btn_logout = (Button)view.findViewById(R.id.btn_logout);
        Button btn_leave = (Button)view.findViewById(R.id.btn_leave);

        iv_label_email.setLayoutParams(LLCommon.Get().ResizeDP(getContext(),iv_label_email.getLayoutParams()));
        iv_label_max.setLayoutParams(LLCommon.Get().ResizeDP(getContext(),iv_label_max.getLayoutParams()));
        iv_label_fix.setLayoutParams(LLCommon.Get().ResizeDP(getContext(),iv_label_fix.getLayoutParams()));
        iv_label_pw.setLayoutParams(LLCommon.Get().ResizeDP(getContext(),iv_label_pw.getLayoutParams()));
        iv_label_push.setLayoutParams(LLCommon.Get().ResizeDP(getContext(),iv_label_push.getLayoutParams()));

        btn_logout.setLayoutParams(LLCommon.Get().ResizeDP(getContext(),btn_logout.getLayoutParams()));
        btn_leave.setLayoutParams(LLCommon.Get().ResizeDP(getContext(),btn_leave.getLayoutParams()));
    }

    View.OnClickListener listener_click_iv = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            Intent intent = null;
            switch (v.getId()){
                case R.id.iv_label_email:
                    intent = new Intent(getContext(), SettingEmailActivity.class);
                    startActivity(intent);
                    break;
                case R.id.iv_label_max:
                    intent = new Intent(getContext(), NumberPopupActivity.class);
                    intent.putExtra("min",5);
                    intent.putExtra("max",100);
                    intent.putExtra("reverse",false);
                    startActivityForResult(intent,MAX);
                    break;
                case R.id.iv_label_fix:
                    intent = new Intent(getContext(), NumberPopupActivity.class);
                    intent.putExtra("min",0);
                    intent.putExtra("max",45);
                    intent.putExtra("reverse",false);
                    startActivityForResult(intent,FIX);
                    break;
                case R.id.iv_label_pw:
                    intent = new Intent(getContext(), SettingPwActivity.class);
                    startActivity(intent);
                    break;
                case R.id.iv_label_push:
                    intent = new Intent(getContext(), SettingPushActivity.class);
                    startActivity(intent);
                    break;
            }
        }
    };
    View.OnTouchListener listener_touch_iv = new View.OnTouchListener() {
        @Override
        public boolean onTouch(View v, MotionEvent event) {
            // TODO Auto-generated method stub
            ImageView view = (ImageView)v;
            if(event.getAction() == MotionEvent.ACTION_DOWN){
                view.getBackground().setColorFilter(0x30000000, PorterDuff.Mode.SRC_OVER);
            }
            else if(event.getAction() == MotionEvent.ACTION_UP){
                view.getBackground().setColorFilter(0x00000000, PorterDuff.Mode.SRC_OVER);
            }
            else if(event.getAction() == MotionEvent.ACTION_CANCEL){
                view.getBackground().setColorFilter(0x00000000, PorterDuff.Mode.SRC_OVER);
                view.invalidate();
            }
            return false;
        }
    };
    View.OnClickListener listener_click_btn = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            switch (v.getId()){
                case R.id.btn_logout:
                    LLSharedPreferences.LL(getContext()).putSaveSession("");
                    if(refresh!=null)
                        refresh.onRefresh();
                    break;
                case R.id.btn_leave:
                    LLAlert.Get(getContext()).CancelAlert("모든 정보가 삭제됩니다.\n정말로 탈퇴하시겠습니까?", new LLAlert.DialogOkCancelListener() {
                        @Override
                        public void DialogOkCancelListener() {
                            String token = LLSharedPreferences.LL(getContext()).getSaveSession();
                            if(LLCommon.Get().CheckExpired(token) == false) {
                                LeaveMember(token);
                            }
                        }
                    });
                    break;
            }
        }
    };
    View.OnTouchListener listener_touch_btn =  new View.OnTouchListener() {
        @Override
        public boolean onTouch(View v, MotionEvent event) {
            Button view = (Button)v;
            if(event.getAction() == MotionEvent.ACTION_DOWN){
                view.getBackground().setColorFilter(0x50000000, PorterDuff.Mode.SRC_OVER);
                view.invalidate();
            }
            else if(event.getAction() == MotionEvent.ACTION_UP){
                view.getBackground().setColorFilter(0x00000000, PorterDuff.Mode.SRC_OVER);
                view.invalidate();
            }
            else if(event.getAction() == MotionEvent.ACTION_CANCEL){
                view.getBackground().setColorFilter(0x00000000, PorterDuff.Mode.SRC_OVER);
                view.invalidate();
            }
            return false;
        }
    };
    private void GetUserInfo(String key){
        try{
            Loading(true);
            ApiAsnycTask api = new ApiAsnycTask(getString(R.string.api_url)+"user/GetUserInfor",key);
            api.execute();
            api.finish = new ApiAsnycTask.Finish() {
                @Override
                public void Finish(String json) {
                    Gson g = new Gson();
                    ApiResult<UserInfo> result = g.fromJson(json, new TypeToken<ApiResult<UserInfo>>(){}.getType());
                    if(result != null){
                        if(result.success){
                            userInfo = result.result;
                            TextView tv_email = (TextView)rootView.findViewById(R.id.tv_email);
                            TextView tv_max = (TextView)rootView.findViewById(R.id.tv_max);
                            TextView tv_fix = (TextView)rootView.findViewById(R.id.tv_fix);
                            tv_email.setText(userInfo.emailAddress);
                            tv_max.setText(userInfo.maxSelectNumber);
                            ArrayList<Integer> list_digit = new ArrayList<>();
                            list_digit.add(Integer.parseInt(userInfo.digit1));
                            list_digit.add(Integer.parseInt(userInfo.digit2));
                            list_digit.add(Integer.parseInt(userInfo.digit3));
                            Collections.sort(list_digit);
                            String fix="-";
                            for(Integer digit : list_digit){
                                if(digit !=0){
                                  if(fix.equals("-"))
                                      fix =digit.toString();
                                  else
                                      fix +=", "+digit.toString();
                                }
                            }
                            tv_fix.setText(fix);
                        }
                        else{
                            LLAlert.Get(getContext()).Alert(result.message);
                        }
                    }
                    Loading(false);
                }
            };
        }
        catch (Exception ex){

        }
    }
    private void UpdateUserInfo(String key, String name, String max, String digit1, String digit2, String digit3){
        try{
            Loading(true);
            HashMap<String, String> parameter = new HashMap<>();
            parameter.put("login_name",name);
            parameter.put("max_select_number",max);
            parameter.put("digit1", digit1);
            parameter.put("digit2", digit2);
            parameter.put("digit3", digit3);
            ApiAsnycTask api = new ApiAsnycTask(getString(R.string.api_url)+"user/UpdateUserInfor",key);
            api.execute(parameter);
            api.finish = new ApiAsnycTask.Finish() {
                @Override
                public void Finish(String json) {
                    Gson g = new Gson();
                    ApiResult<String> result = g.fromJson(json, new TypeToken<ApiResult<String>>(){}.getType());
                    if(result != null){
                        if(result.success){
                            LLAlert.Get(getContext()).Alert("변경되었습니다.", new LLAlert.DialogOkCancelListener() {
                                @Override
                                public void DialogOkCancelListener() {
                                    Init();
                                }
                            });
                        }
                        else{
                            LLAlert.Get(getContext()).Alert(result.message);
                        }
                    }
                    Loading(false);
                }
            };
        }
        catch (Exception ex){

        }
    }
    private void GetCount(String key){
        try{
            ApiAsnycTask api = new ApiAsnycTask(getString(R.string.api_url)+"notify/GetCount",key);
            api.execute();
            api.finish = new ApiAsnycTask.Finish() {
                @Override
                public void Finish(String json) {
                    Gson g = new Gson();
                    ApiResult<String> result = g.fromJson(json, new TypeToken<ApiResult<String>>(){}.getType());
                    if(result != null){
                        if(result.success){
                            TextView tv_push = (TextView)rootView.findViewById(R.id.tv_push);
                            tv_push.setText(result.result);
                        }
                        else{
                            LLAlert.Get(getContext()).Alert(result.message);
                        }
                    }
                }
            };
        }
        catch (Exception ex){

        }
    }
    private void LeaveMember(String key){
        try{
            Loading(true);
            ApiAsnycTask api = new ApiAsnycTask(getString(R.string.api_url)+"user/LeaveMember",key);
            api.execute();
            api.finish = new ApiAsnycTask.Finish() {
                @Override
                public void Finish(String json) {
                    Gson g = new Gson();
                    ApiResult<String> result = g.fromJson(json, new TypeToken<ApiResult<String>>(){}.getType());
                    if(result != null){
                        if(result.success){
                            LLSharedPreferences.LL(getContext()).putSaveSession("");
                            if(refresh!=null)
                                refresh.onRefresh();
                        }
                        else{
                            LLAlert.Get(getContext()).Alert(result.message);
                        }
                    }
                    Loading(false);
                }
            };
        }
        catch (Exception ex){

        }
    }
    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        Loading(true);
        switch (requestCode) {
            case MAX:
                if (resultCode == Activity.RESULT_OK && data != null) {
                    userInfo.maxSelectNumber = String.valueOf(data.getIntExtra("number",0));
                    UpdateUserInfo(LLSharedPreferences.LL(getContext()).getSaveSession(),userInfo.loginName,userInfo.maxSelectNumber,userInfo.digit1,userInfo.digit2,userInfo.digit3);
                }
                break;
            case FIX:
                if (resultCode == Activity.RESULT_OK && data != null) {
                    String sequence = String.valueOf(data.getIntExtra("number",0));
                    if(sequence.equals("0")){
                            userInfo.digit3 = sequence;
                            userInfo.digit2 = sequence;
                            userInfo.digit1 = sequence;
                    }else {
                        if(!(userInfo.digit1.equals(sequence) || userInfo.digit2.equals(sequence) || userInfo.digit3.equals(sequence))){
                            if(userInfo.digit1.equals("0") == true){
                                userInfo.digit1 = String.valueOf(sequence);
                            }
                            else if (userInfo.digit2.equals("0") == true){
                                userInfo.digit2 = String.valueOf(sequence);
                            }
                            else{
                                userInfo.digit3 = String.valueOf(sequence);
                            }
                        }
                    }
                    UpdateUserInfo(LLSharedPreferences.LL(getContext()).getSaveSession(),userInfo.loginName,userInfo.maxSelectNumber,userInfo.digit1,userInfo.digit2,userInfo.digit3);
                }
                break;
        }
    }

    private void Loading(boolean view){
        Button btn_loading = (Button)rootView.findViewById(R.id.btn_loading);
        ProgressBar pb_loading = (ProgressBar)rootView.findViewById(R.id.pb_loading);

        if(view){
            btn_loading.setVisibility(View.VISIBLE);
            pb_loading.setVisibility(View.VISIBLE);
        }
        else{
            btn_loading.setVisibility(View.INVISIBLE);
            pb_loading.setVisibility(View.INVISIBLE);
        }
    }
}
